client/components/jetpack/daily-backup-status/index.jsx

```diff
@@ -86,6 +98,31 @@ const DailyBackupStatus = ( {
 		? moment( backupPreviouslyInProgress.current.period * 1000 )
 		: undefined;
 
+	// State for tracking the last backup that was in progress.
+	const [ lastBackup, setLastBackup ] = useState( null );
+
+	// Effect for handling backup updates and Activity Log fetching intervals.
+	useEffect( () => {
+		// Set lastBackup on initial load or clear interval if backup's rewindId changes.
+		if ( backup && ! lastBackup ) {
+			setLastBackup( backup );
+		} else if ( backup && lastBackup && backup.rewindId !== lastBackup.rewindId ) {
+			backupPreviouslyInProgress.current = null;
+			clearActivityLogInterval();
+			setLastBackup( backup );
```

Comment:
In this scenario, we already have the new backup, so we can set it as the last backup to fix https://github.com/Automattic/wp-calypso/pull/89751#issuecomment-2071084187.